src/libcephfs.cc

```diff
@@ -2127,13 +2127,13 @@ extern "C" void ceph_ll_readv_writev_release(void *release_data)
   delete buf;
 }
 
-extern "C" void ceph_ll_readv_writev(class ceph_mount_info *cmount,
+extern "C" int64_t ceph_ll_readv_writev(class ceph_mount_info *cmount,
 				     struct ceph_ll_io_info *io_info)
 {
   ceph_ll_readv_writev_buffer *buf = new ceph_ll_readv_writev_buffer;
 
   if (!io_info->write && io_info->zerocopy) {
-    ceph_assert(io_info->iovcnt == 1);
```

Comment:
We need this test for io_info->iovcnt in the if since we are no longer doing it in assert

So the if test should be:

// a zero copy read MUST provide a length 1 iovec (where the buffer is a nullptr and the length is the requested read length
if (!io_info->write && io_info->zerocopy && (io_info->iovcnt != 1) {

Reply:
i dont know how i missed it, we talked about it on slack. fixing.